.gitignore

```
@@ -45,3 +45,5 @@ nlohmann_json.spdx
 
 # Bazel-related
 MODULE.bazel.lock
+enum.log
+tidy.log
```

Why did you edit .gitignore?

---

Thanks for catching that.
I added tidy.log and the matching entry in .gitignore only for local debugging it doesn’t belong in the PR. I’ll drop the file and revert the .gitignore change, then force‑push an updated commit with the proper DCO trailer.

---

I still have not understood why this PR tries to achieve.

---

The CI started failing because GitHub’s runners switched to clang‑tidy 18.
That version enables a new rule (cppcoreguidelines‑use‑enum‑class) which floods the build with warnings and turns the check red.

All I’m trying to do in this PR is pin the workflow back to clang‑tidy 17 so the checks pass again nothing else.

I accidentally pushed tidy.log and the .gitignore tweak sorry about that. if you want ,  I’m removing both and will re‑push with a clean, signed‑off commit.

---

The last successful CI run (https://github.com/nlohmann/json/actions/runs/16006669329/job/45154866131?pr=4587) is already using Clang-Tidy 21.0.0.

---

One tidy commit:
switched the unscoped enum to enum class
cleaned out the stray log / .gitignore change
still on clang‑tidy 21, signed off

Could you pull the branch and see if it’s good now?
Happy to jump on any other C++ clean‑ups  you need next.

Thanks!

---

I'm already addressing these in #4871.

---

In #4871, I fixed all code-related errors. What's missing is related to changed CI images.

---

Got it thanks for clarifying!
I see #4871 already handles the clang-tidy/code fixes.
I’ll close this PR and open or help on a follow-up that updates the CI workflow images so the pipeline goes green again.
Let me know if you’d prefer me to tackle a specific job matrix or leave it with you.

---

I think I get the Ubuntu stuff to work soon. For Windows, I could use some help. Best case scenario would be to keep all existing jobs running, but I guess that some are bound to the deprecated/removed images. It would be great if you could have a look.

Best start from the develop branch (not #4871 where I already deleted some jobs) and just ignore the errors in Ubuntu for the time being.

---

I created #4875 for this. It would be great if you could take a look!